integration/flink/examples/flink2-test-apps/src/main/java/io/openlineage/flink/FlinkSqlApplication.java

```diff
@@ -0,0 +1,103 @@
+/*
+/* Copyright 2018-2025 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.flink;
+
+import static io.openlineage.flink.StreamEnvironment.setupEnv;
+
+import java.util.concurrent.TimeUnit;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.connectors.kafka.table.KafkaDynamicTableFactory;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.util.ParameterTool;
+import org.apache.kafka.common.errors.TimeoutException;
+
+public class FlinkSqlApplication {
+
+  public static void main(String[] args) throws Exception {
+    ParameterTool parameters = ParameterTool.fromArgs(args);
+    StreamExecutionEnvironment env = setupEnv(args);
+    StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+    // ---------- Produce an event time stream into Kafka -------------------
+    String groupId = "testSql";
+    String bootstraps = parameters.getRequired("bootstraps");
+
+    final String createInputTable =
+        String.format(
+            "create table kafka_input (\n"
+                + "  `computed-price` as price + 1.0,\n"
+                + "  price decimal(38, 18),\n"
+                + "  currency string,\n"
+                + "  log_date date,\n"
+                + "  log_time time(3),\n"
+                + "  log_ts timestamp(3),\n"
+                + "  ts as log_ts + INTERVAL '1' SECOND,\n"
+                + "  watermark for ts as ts\n"
+                + ") comment 'My Complex Table' with (\n"
+                + "  'connector' = '%s',\n"
+                + "  'topic' = '%s',\n"
+                + "  'properties.bootstrap.servers' = '%s',\n"
+                + "  'properties.group.id' = '%s',\n"
+                + "  'scan.startup.mode' = 'earliest-offset',\n"
+                + "  'format' = 'json'\n"
+                + ")",
```

Comment:
nit: Flink 2 works with Java 17, might be good to use text blocks feature for this.

Reply:
you're right, but I think we should instead use java 17 for the whole flink compile & test process with java compatibility flags set. Otherwise, mixing different Java's adds more burden than the benefit of text blocks.